### PR TITLE
Fixes #20336 - fix template lookup on os creation

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: app/models/katello/concerns/operatingsystem_extensions.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def assign_templates!
               provisioning_template_name = Setting["katello_default_#{kind.name}"]
             end
 
-            if (template = ProvisioningTemplate.find_by_name(provisioning_template_name))
+            if (template = ProvisioningTemplate.unscoped.find_by_name(provisioning_template_name))
```
Comment (bbuckingham) [inline app/models/katello/concerns/operatingsystem_extensions.rb]: After this change, it does appear to associate the templates, which is good. :)

Do we need also do the unscoped query for any other foreman resources used by this flow?  (e.g. Ptable, ...)

Comment (bbuckingham) [inline app/models/katello/concerns/operatingsystem_extensions.rb]: Confirmed with testing that at least Ptable also needs to be unscoped.  Others to check may be OsDefaultTemplate and TemplateKind.

Comment (jlsherrill) [inline app/models/katello/concerns/operatingsystem_extensions.rb]: Good catch on PTable, i don't think OSDefaultTemplate or TemplateKind are restricted by permissions, so I don't think they are necessary

Comment (jlsherrill) [inline app/models/katello/concerns/operatingsystem_extensions.rb]: updated!

**Pull request merged**